packages/mobile/src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -0,0 +1,310 @@
+import KeyboardSpacer from '@celo/react-components/components/KeyboardSpacer'
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import variables from '@celo/react-components/styles/variables'
+import { parseInputAmount } from '@celo/utils/lib/parsing'
+import { RouteProp } from '@react-navigation/core'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import React, { useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { Platform, StyleSheet, Text, TextInput, View } from 'react-native'
+import { getNumberFormatSettings } from 'react-native-localize'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { cUsdDailyLimitSelector } from 'src/account/selectors'
+import BackButton from 'src/components/BackButton'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import Dialog from 'src/components/Dialog'
+import LineItemRow from 'src/components/LineItemRow'
+import { CELO_SUPPORT_EMAIL_ADDRESS, DOLLAR_ADD_FUNDS_MIN_AMOUNT } from 'src/config'
+import { fetchExchangeRate } from 'src/exchange/actions'
+import { exchangeRatePairSelector } from 'src/exchange/reducer'
+import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
+import i18n, { Namespaces } from 'src/i18n'
+import { LocalCurrencySymbol } from 'src/localCurrency/consts'
+import {
+  convertDollarsToLocalAmount,
+  convertDollarsToMaxSupportedPrecision,
+  convertLocalAmountToDollars,
+} from 'src/localCurrency/convert'
+import { useLocalCurrencyCode } from 'src/localCurrency/hooks'
+import { getLocalCurrencyExchangeRate } from 'src/localCurrency/selectors'
+import { emptyHeader, HeaderTitleWithBalance } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import { getRateForMakerToken, goldToDollarAmount } from 'src/utils/currencyExchange'
+
+const { decimalSeparator } = getNumberFormatSettings()
+
+type RouteProps = StackScreenProps<StackParamList, Screens.FiatExchangeAmount>
+
+type Props = RouteProps
+
+const oneDollarAmount = {
+  value: new BigNumber('1'),
+  currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
+}
+
+const useDollarAmount = (currency: CURRENCY_ENUM, amount: BigNumber) => {
+  const localExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencyCode = useLocalCurrencyCode()
+  const exchangeRatePair = useSelector(exchangeRatePairSelector)
+
+  if (currency === CURRENCY_ENUM.DOLLAR) {
+    return convertDollarsToMaxSupportedPrecision(
+      (!amount.isNaN() &&
+        convertLocalAmountToDollars(amount, localCurrencyCode ? localExchangeRate : 1)) ||
+        new BigNumber('0')
```

Comment:
Nit: can we make this more legible? It took me a second to understand what was happening because it's super dense in this format.